fabulaws/library/wsgiautoscale/api.py

```diff
@@ -466,7 +466,12 @@ def _new(deployment, environment, role, avail_zone=None, count=1, **kwargs):
                      tags=tags, volume_size=vol_size, volume_type=vol_type,
                      deploy_user=env.deploy_user, security_groups=sec_grps,
                      **extra_args)
-        server.setup()
+        try:
+            server.setup()
+        except:
+            if terminate_on_failure:
+                server.terminate()
```

Comment:
I was assuming the error would propagate up via `raise` to whatever logging/handling is already in place. E.g., I manually killed an `apt-get` command that was in the middle of running on a server and got this:

	Restoring context
	Setting up context
	[ec2-54-161-221-155.compute-1.amazonaws.com] sudo: apt-get -qq -y install ntp
	[ec2-54-161-221-155.compute-1.amazonaws.com] out: E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)
	[ec2-54-161-221-155.compute-1.amazonaws.com] out: E: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?
	[ec2-54-161-221-155.compute-1.amazonaws.com] out: 


	Fatal error: sudo() received nonzero return code 100 while executing!

	Requested: apt-get -qq -y install ntp
	Executed: sudo -S -p 'sudo password:'  /bin/bash -l -c "apt-get -qq -y install ntp"

	Aborting.
	Restoring context
	Server creation failed; retrying...
	The terminate, tags, and placement arguments have no effect when instance_id is set.
	Creating EC2 instances
	Waiting for instance to enter "running" state...
	Waiting for SSH daemon to launch...

Or do you think it would still be helpful to log the exception and/or the fact that we're terminating the instance here? For the most part the exception will probably be the one I define as `abort_exception`, but it might still contain some useful information.